presto-bytecode/src/main/java/com/facebook/presto/bytecode/DynamicClassLoader.java

```diff
@@ -30,16 +31,27 @@
 {
     private final ConcurrentMap<String, byte[]> pendingClasses = new ConcurrentHashMap<>();
     private final Map<Long, MethodHandle> callSiteBindings;
+    private final Optional<ClassLoader> secondaryParent;
 
     public DynamicClassLoader(ClassLoader parentClassLoader)
     {
         this(parentClassLoader, ImmutableMap.of());
     }
 
+    // TODO: this is a hack that should be removed
+    @Deprecated
+    public DynamicClassLoader(ClassLoader parentClassLoader, ClassLoader secondaryParent)
```

Comment:
I would name these, `firstParentClassLoader` and `secondParentClassLoader`.  Then I would check them in that order.  If you want, the current order in this PR, just swap the order in the callsite.